### PR TITLE
Cleanup available settings, stage for proper options handling

### PR DESCRIPTION
**Pull request by jarolrod**
based on bitcoin-core/gui-qml#156 and bitcoin-core/gui-qml#157
This primarily cleans up settings/options that cannot be modified by the OptionsModel we will use; thus staging the gui for proper options handling. 

Introduces a static backup for the storage lottie element. The SVG was provided by @GBKS, and the png generated with inkscape. The resulting png was then run through the optimize python script, the output of the script is as follows:
```
optimizing storage-dark.png... done
optimizing storage-light.png... done
summary:
+++++++++++++++++
storage-dark.png
  size diff from: 10942 to: 8412
  old sha256: ff0937aabffa949a241df90ff3a61d9e3cb35a836a11e497718196097b0e68fb
  new sha256: 8a3c195ee3b44624c89ab648020005a0e955b7e6086a89c966de0547f337b98b

storage-light.png
  size diff from: 11205 to: 8693
  old sha256: f7e707cd997f8acf3ac273f6680611f420375ed40cdbacb0a18e2003d49496d4
  new sha256: 38ad2e515a55980d01cfeaeee0bb7264005a739e65fc57f4bb99cdb92e39f3c6

completed. Checksum stable: False. Total reduction: 5042 bytes
```

# Master
| Developer Options | Storage Location | Storage | 
| ------------------------- | ----------------------- | ----------- | 
| master-2 | master-3 | master-4 | 

| Storage Settings | Connection | Connection Settings |
| --------------------- | --------------- | --------------------------- |
| master-5 | master-6 | master-7 |

# PR
| Developer Options | Storage Location | Storage | 
| ------------------------- | ----------------------- | ----------- | 
| newdev-screen | pr-3 | pr-4 |

| Storage Settings | Connection | Connection Settings |
| --------------------- | --------------- | --------------------------- |
| pr-5 | new-ibd-screen | newcon-screen |

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/158)
[Intel macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/158)
[Apple Silicon macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/158)
[ARM64 Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/158)

### CODE REVIEW

**Comment by jarolrod** [comment]
updated from 674117d to be2c533

changes: rebased over changes on master, dropped unneeded PR dependencies

**Comment by jarolrod** [comment]
converted to draft until we are able to have a discussion on the bitcoin core app call with designers

**Comment by jarolrod** [comment]
updated from be2c533 to 16fa2d4

Changes: based on new PR's to not conflict, used correct storage image

**Comment by GBKS** [comment]
Awesome progress.

File: src/qml/pages/onboarding/onboarding06a.qml [hunk 1]
```diff
@@ -24,22 +24,27 @@ Page {
...
+            header: qsTr("Starting initial download")
+            headerMargin: 30
+            description: qsTr("The application will connect to the Bitcoin network and start downloading and verifying transactions.\n\nThis may take several hours, or even days, based on your connection.")
+            descriptionMargin: 20
         }
         TextButton {
             Layout.alignment: Qt.AlignCenter
             Layout.topMargin: 30
-            text: "Detailed Settings"
+            text: "Connection Settings"
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qml/pages/onboarding/onboarding06a.qml]: addressed in 0f969b8 , thanks!

**Comment by jarolrod** [comment]
updated from 16fa2d4 to 0f969b8 

changes: rebased over main, moved `dbcache` and `scripthreads` settings from Connection settings to Developer settings, corrected a [capitalization error](https://github.com/bitcoin-core/gui-qml/pull/158#discussion_r947993292)

File: src/qml/components/DeveloperOptions.qml [hunk 2]
```diff
@@ -19,14 +19,16 @@ ColumnLayout {
...
-        header: qsTr("Storage limit")
-        actionItem: OptionSwitch {}
+        header: qsTr("Database cache size")
+        actionItem: ValueInput {
+            description: ("450 MiB")
+        }
     }
     Setting {
         Layout.fillWidth: true
-        header: qsTr("Network")
```
Comment (shaavan) [inline src/qml/components/DeveloperOptions.qml:27]: 4. I was able to verify the non-modifiability by the optionsModel of all the removed settings except this one (”Network”), so could you provide some details on the file responsible for handling the chain switching?

File: src/qml/components/StorageSettings.qml [hunk 3]
```diff
@@ -27,13 +27,5 @@ ColumnLayout {
         actionItem: ValueInput {
             description: qsTr("c://.../data")
         }
```
Comment (shaavan) [inline src/qml/components/StorageSettings.qml:29]: 5. The storageLocation.qml component allows selecting the location for data storage. So providing the same option in the StorageSettings during the onboarding process seems redundant.

**Review by shaavan — COMMENTED** [review]
Concept ACK

Fantastic work, @jarolrod! 

I have some points to discuss/add that can further improve this clean-up.

1. The ConnectionOptions.qml file is now irrelevant and can be removed.

2. In StorageSettings.qml:

```qml
Setting {
        Layout.fillWidth: true
        header: qsTr("Store Recent blocks only")
        actionItem: OptionSwitch {}
    }
    Setting {
        Layout.fillWidth: true
        header: qsTr("Storage limit")
        actionItem: ValueInput {
            description: qsTr("2 GB")
        }
    }
```
Doesn’t giving a "storage limit" imply that only the recent block will be stored? So is providing a separate option for "store only recent blocks" correct?

File: src/qml/components/ConnectionSettings.qml [hunk 4]
```diff
@@ -11,42 +11,24 @@ ColumnLayout {
...
-        header: qsTr("Enable listening")
-        description: qsTr("Increases data usage")
+        header: qsTr("Map port using UPnP")
         actionItem: OptionSwitch {}
     }
     Setting {
         Layout.fillWidth: true
-        header: qsTr("Blocks Only")
-        description: qsTr("Reduces data usage.")
+        header: qsTr("Map port using NAT-PMP")
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qml/components/ConnectionSettings.qml:25]: yes, they can be enabled at the same time

File: src/qml/components/DeveloperOptions.qml (hunk 2 above)
Comment (jarolrod) [reply src/qml/components/DeveloperOptions.qml:27]: there is no file responsible for handling chain switching, that isn't something we can do as of now

File: src/qml/components/StorageSettings.qml (hunk 3 above)
Comment (jarolrod) [reply src/qml/components/StorageSettings.qml:29]: Yes as it should be, the first screen you reference is the easy path settings, the next is a detailed settings screen.

**Comment by jarolrod** [comment]
@shaavan 
> The ConnectionOptions.qml file is now irrelevant and can be removed.

Yes, but that isn't going to be removed until there's absolutely no need for it in the future. This is useful when porting to mobile

> Doesn’t giving a "storage limit" imply that only the recent block will be stored? So is providing a separate option for "store only recent blocks" correct?

This PR doesn't intend to solve all issues around the workings of the settings themselves. Just cleaning up from fake settings to the real ones. This is correct as is. The UI issue here is that the value input field for the prune size should be disabled until the prune button is enabled. The UI pattern to use there is being discussed with designers.

File: src/qml/components/StorageSettings.qml (hunk 3 above)
Comment (shaavan) [reply src/qml/components/StorageSettings.qml:29]: @jarolrod

Thanks for providing the clarification. But I still have some points to discuss.

The **storage location** page provides two options: The **Default Directory** and the **Custom Directory**. And as per my understanding, clicking on the **Custom Directory** will open the file manager to allow the user to choose the directories location. So having the exact same option in the storage setting still seems redundant to me.

Also, the Storage Settings page deals with the size-related factors regarding storage, so having a storage location option seems out of place when we have a dedicated onboarding page for the storage location selection.

I think it is better to keep the functionality of each page mutually exclusive and avoid overlap in their usage.

These are my thoughts on this matter, but I agree with your reasoning [here](https://github.com/bitcoin-core/gui-qml/pull/158#issuecomment-1223512830), and removing UI-based redundancies is out of the scope of this PR, so we can further brainstorm on this with the designer as well.

**Comment by shaavan** [comment]
> Yes, but that isn't going to be removed until there's absolutely no need for it in the future. This is useful when porting to mobile

That seems reasonable.

> The UI issue is that the value input field for the prune size should be disabled until the prune button is enabled.

Maybe this can be solved by adding an "is_enable" property to the setting control, which displays the content of settings but makes them unclickable. But I have to further experiment with this idea to see if it will work.

**Comment by jarolrod** [comment]
> Maybe this can be solved by adding an "is_enable" property to the setting control, which displays the content of settings but makes them unclickable. But I have to further experiment with this idea to see if it will work.

Yeah it's straight forward to do this with binding, just something to figure out on the next iterative step, when there is actually options handling going on

**Comment by jarolrod** [comment]
updated from[ 0f969b8 ](https://github.com/bitcoin-core/gui-qml/commit/0f969b8df5bf02e150198210918ba2b48b772aab)to[ d81c43d](https://github.com/bitcoin-core/gui-qml/commit/d81c43dd6f1fa901760f8d9d6914722522d5aa8a)

changes: rebased over master and fixed conflicts

**Review by hebasto — APPROVED** [review]
ACK d81c43dd6f1fa901760f8d9d6914722522d5aa8a, I have reviewed the code and it looks OK, I agree it can be merged.

File: src/qml/components/StorageOptions.qml [hunk 5]
```diff
@@ -15,7 +15,7 @@ ColumnLayout {
     OptionButton {
         ButtonGroup.group: group
         text: qsTr("Reduce storage")
-        description: qsTr("Uses about 75GB.")
+        description: qsTr("Uses about 2GB.")
```
Comment (hebasto) [inline src/qml/components/StorageOptions.qml:18]: In other places we use a space to separate a unit from a value.

**Pull request merged**